src/datasources/db/v2/database-migrator.service.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import { DatabaseMigrator } from '@/datasources/db/v2/database-migrator.service';
+import {
+  LoggingService,
+  type ILoggingService,
+} from '@/logging/logging.interface';
+
+describe('PostgresDatabaseService', () => {
+  let postgresqlService: PostgresDatabaseService;
+  let databaseMigratorService: DatabaseMigrator;
+  let loggingService: ILoggingService;
+
+  beforeEach(async () => {
+    // We should not require an SSL connection if using the database provided
+    // by GitHub actions
+    const isCIContext = process.env.CI?.toLowerCase() === 'true';
+    const baseConfiguration = configuration();
+    const testConfiguration: typeof configuration = () => ({
+      ...baseConfiguration,
+      db: {
+        ...baseConfiguration.db,
+        connection: {
+          postgres: {
+            ...baseConfiguration.db.connection.postgres,
+            ssl: {
+              ...baseConfiguration.db.connection.postgres.ssl,
+              enabled: !isCIContext,
+              caPath: join(__dirname, '../../../../db_config/test/server.crt'),
+            },
+          },
+        },
+      },
+    });
+
+    const moduleRef: TestingModule = await Test.createTestingModule({
+      imports: [
+        TypeOrmModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: async (configService: ConfigService) => {
+            const typeormConfig = await configService.getOrThrow('db.orm');
+            const postgresConfigObject = postgresConfig(
+              await configService.getOrThrow('db.connection.postgres'),
+            );
+
+            return {
+              ...typeormConfig,
+              ...postgresConfigObject,
+              ...{
+                migrations: ['dist/migrations/test/*.js'],
+              },
+            };
+          },
+          inject: [ConfigService],
+        }),
+        TestLoggingModule,
+        ConfigurationModule.register(testConfiguration),
+        ConfigModule,
+      ],
+      providers: [PostgresDatabaseService, DatabaseMigrator],
+    }).compile();
+
+    postgresqlService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+    databaseMigratorService = moduleRef.get<DatabaseMigrator>(DatabaseMigrator);
+    loggingService = moduleRef.get<ILoggingService>(LoggingService);
+  });
+
+  afterAll(async () => {
+    await postgresqlService.destroyDatabaseConnection();
+  });
+
+  describe('migrate()', () => {
+    it('Should log the start and end of the migration process', async () => {
+      jest.spyOn(loggingService, 'info');
+
+      await databaseMigratorService.migrate();
+
+      expect(loggingService.info).toHaveBeenCalledWith(
+        'Migrations: Running...',
+      );
+      expect(loggingService.info).toHaveBeenCalledWith('Migrations: Finished.');
```

Comment:
Nit: I'd also check the number of calls.

General remark: this is an interesting edge case in the codebase. Normally we'd use a mock service that has jest mocks in place but the `TestLoggingModule` doesn't.

Reply:
Fixed this one and Added more unit tests https://github.com/safe-global/safe-client-gateway/pull/1988/commits/f7da627dc0505039d12c1605cc646aeda59c9c68